src/store/themeSlice.js

```diff
@@ -0,0 +1,22 @@
+import { createSlice } from '@reduxjs/toolkit';
+import { getItemFromLocalStorage, setItemInLocalStorage } from '../utils/local-storage-utils';
+
+const initialState = {
+  darkMode: getItemFromLocalStorage('darkMode'),
+};
+
+const themeSlice = createSlice({
+  name: 'theme',
```

Comment:
Why do we need to keep the theme state in redux level?

Reply:
So that it is easier to fetch the current value to apply for all the components in the app

Reply:
Please do let me know your thoughts

Reply:
Its debatable...however I would prefer using context over redux for this use case.
We can keep it in redux though. Can you explore on what data needs to be stored in redux and can we use context if we are using redux?